go-controller/pkg/clustermanager/node/subnet_allocator.go

```diff
@@ -359,6 +360,18 @@ func (snr *subnetAllocatorRange) markAllocatedNetwork(owner string, network *net
 
 // allocateNetwork returns a new subnet, or nil if the range is full
 func (snr *subnetAllocatorRange) allocateNetwork(owner string) *net.IPNet {
+	// Return an already allocated subnet instead of creating a new one if a
+	// combination of subnet and node name already exists in the cache
+	for nodeSubnet, nodeName := range snr.allocMap {
+		if nodeName == owner {
```

Comment:
let's say owner = nodeA but I have a v4 and v6 range for this nodeSubnet, is that handled correctly? IIRC this subnetAllocatorRange runs a separate instances for each family?

Reply:
I guess a nicer way to do this is to make owner a thing inside the subnetAllocatorRange but I am ok with this change as of now

Reply:
Yes, subnetAllocatorRange is different for different IP families.
https://github.com/arghosh93/ovn-kubernetes/blob/master/go-controller/pkg/clustermanager/node/subnet_allocator.go#L155-L185